src/main/java/org/sagebionetworks/bridge/udd/synapse/SynapseDownloadFromTableTask.java

```diff
@@ -0,0 +1,421 @@
+        } catch (RuntimeException ex) {
+            // Cleanup files. No need to leave garbage behind.
+            cleanupFiles();
+            throw new AsyncTaskExecutionException(ex);
+        }
+    }
+
+    /**
+     * Queries a Synapse table based on the params and downloads the result as a CSV. This method reads all params
+     * (except schema) from {@link SynapseDownloadFromTableParameters} to generate the query and writes the resulting
+     * CSV to {@link SynapseDownloadFromTableContext#setCsvFile}.
+     */
+    private void downloadCsv() {
+        String synapseTableId = params.getSynapseTableId();
+        File csvFile = fileHelper.newFile(params.getTempDir(), params.getSchema().getKey().toString() + ".csv");
+        String csvFilePath = csvFile.getAbsolutePath();
+
+        Stopwatch downloadCsvStopwatch = Stopwatch.createStarted();
+        try {
+            String query = String.format(QUERY_TEMPLATE, synapseTableId, params.getHealthCode(), params.getStartDate(),
+                    params.getEndDate());
+            String csvFileHandleId = synapseHelper.generateFileHandleFromTableQuery(query, synapseTableId);
+            synapseHelper.downloadFileHandle(csvFileHandleId, csvFile);
+            ctx.setCsvFile(csvFile);
+        } catch (AsyncTimeoutException | SynapseException ex) {
+            throw new AsyncTaskExecutionException("Error downloading synapse table " + synapseTableId + " to file " +
+                    csvFilePath + ": " + ex.getMessage(), ex);
+        } finally {
+            downloadCsvStopwatch.stop();
+            LOG.info("Downloading from synapse table " + synapseTableId + " to file " + csvFilePath + " took " +
+                    downloadCsvStopwatch.elapsed(TimeUnit.MILLISECONDS) + " ms");
+        }
+    }
+
+    /**
+     * <p>
+     * Sometimes, a Synapse table contains no data for the given user and time range. This method counts the lines in
+     * the file to determine if that is the case. If there aren't at least 2 lines (header row plus at least one data
+     * row), we filter out the file. (The actual filtering is done by the caller. We simply return true if we should
+     * filter.)
+     * </p>
+     * <p>
+     * This method reads from {@link SynapseDownloadFromTableContext#getCsvFile} and doesn't write anything back to
+     * the context.
+     * </p>
+     *
+     * @return true if the file should be filtered because there's no user data
+     */
+    private boolean filterNoDataCsvFiles() {
+        int numLines = 0;
+        try (BufferedReader csvFileReader = fileHelper.getReader(ctx.getCsvFile())) {
+            while (csvFileReader.readLine() != null) {
+                numLines++;
+                if (numLines >= 2) {
+                    // We only need to read 2 lines (1 header, 1 user data). Now that we've read 2 lines, we can
+                    // short-circuit
+                    break;
```

Comment:
Seems like this method could be shorter by returning false here, then cleaning up and returning true after while loop (which, because file was empty, only made it 0-1 loops anyway).